src/net/tcp.rs

```diff
@@ -733,3 +733,60 @@ impl FromRawFd for TcpListener {
         }
     }
 }
+
+/*
+ *
+ * ===== Windows ext =====
+ *
+ */
+
+#[cfg(windows)]
+use std::os::windows::io::{IntoRawSocket, AsRawSocket, FromRawSocket, RawSocket};
+
+#[cfg(windows)]
+impl IntoRawSocket for TcpStream {
+    fn into_raw_socket(self) -> RawSocket {
+        self.sys.into_raw_socket()
+    }
+}
+
+#[cfg(windows)]
+impl AsRawSocket for TcpStream {
+    fn as_raw_socket(&self) -> RawSocket {
+        self.sys.as_raw_socket()
+    }
+}
+
+#[cfg(windows)]
+impl FromRawSocket for TcpStream {
+    unsafe fn from_raw_socket(socket: RawSocket) -> TcpStream {
+        TcpStream {
+            sys: FromRawSocket::from_raw_socket(socket),
+            selector_id: SelectorId::new(),
```

Comment:
I'm not familiar with the Windows side of mio, but is it ok to create a new id here? It doesn't cause double events when calling `TcpStream::from_raw_socket(stream.as_raw_socket())` or something?

Reply:
If you do that you'd actually expect to get double events, or not? You have *two* mio sockets for the same underlying socket, and both should get events presumably.

It's probably not going to work very well though, actions on both sockets will likely trip over each other. `AsRawSocket` only really makes sense to be used to get a handle to the underlying socket to configure things on it, you have to be very careful to not break other users of the same socket as they usually assume unique ownership of it.

Reply:
@sdroege I won't expect to get double events if I convert a `TcpStream` into a `RawSocket`, do some Windows' specific configuration and then convert it back into a `TcpStream`. Atleast on Unix this will work fine and won't generate two events, as only one file descriptor is open (and registered).

Reply:
Your example was using `AsRawSocket`, not `IntoRawSocket`. The former keeps the original mio socket alive, so I would expect to get events on both sockets. You'll have two mio sockets around the same underlying socket.

Reply:
@sdroege Sorry you're right I meant `IntoRawSocket`.